### ISSUE TITLE
[SPIR-V]  C++-style initialization of vectors (`int2{0,1}`) does not work, but compiles.

### ISSUE THREAD

**Issue by nicebyte** [microsoft/DirectXShaderCompiler#7111]
**Description**
C++-style initialization of vectors, e.g. `int2 x{0,1};` appears to generate `OpUndef` in the SPIR-V output.
One effect that i have observed from this is that any comparisons with data from that vector are eval'ed to `false`.

**Steps to Reproduce**
DXC 1.8, with the following shader ([godbolt link](https://godbolt.org/z/ecGPGh5Gf)):

```
// The entry point and target profile are needed to compile this example:
// -T ps_6_6 -E PSMain

struct PSInput
{
    float4 position : SV_Position;
    float4 color    : COLOR0;
    float x : COLOR1;
    float y : COLOR2;
};


float4 PSMain(PSInput input) : SV_Target0
{
  int2 a{2, 1};
  return a.x == 2 ? float4(1.0,1.0,1.0,1.0) : float4(0.0,0.0,0.0,0.0);
}
```

**Actual Behavior**
`OpUndef` in generated spir-v and erratic behavior described above

I'd expect one of two outcomes:
 - this syntax to mean initialization and work same way as in c++
 -  compile error.

**Environment**
- DXC version 1.8
- Host Operating System any

**Comment by damyanp** [comment]
This warning is also emitted:

```
<source>:15:9: warning: effect state block ignored - effect syntax is deprecated. To use braces as an initializer use them with equal signs. [-Weffects-syntax]
  int2 a{2, 1};
```

This should be an error.

@llvm-beanz to file an hlsl-specs issue for figuring out how to deprecate the effects runtime syntax in hlsl202x.
